### PR TITLE
LPS-182671 Clean up subclasses of class BaseModelPermissionChecker | part 1

### PR DESCRIPTION
**Pull request by W2jx**
resend

Previous: https://github.com/liferay-core-infra/liferay-portal/pull/4556

### CODE REVIEW

**Comment by W2jx** [comment]
ci:test:document

**Comment by W2jx** [comment]
ci:test:message-boards

**Comment by W2jx** [comment]
only modify commit message, no need to rerun test

**Comment by Kvale1733** [comment]
@W2jx No regressions found.

**Comment by W2jx** [comment]
no conflicts(6/2 8:25)

File: modules/apps/document-library/document-library-item-selector-web/src/main/java/com/liferay/document/library/item/selector/web/internal/folder/DLFolderItemSelectorView.java [hunk 1]
```diff
@@ -241,6 +243,11 @@ private long _getRepositoryGroupId(long repositoryId) {
 	@Reference
 	private DLAppService _dlAppService;
 
+	@Reference(
+		target = "(model.class.name=com.liferay.portal.kernel.repository.model.Folder)"
+	)
+	private ModelResourcePermission<Folder> _folderModelResourcePermission;
+
 	@Reference
```
Comment (yuhai) [inline modules/apps/document-library/document-library-item-selector-web/src/main/java/com/liferay/document/library/item/selector/web/internal/folder/DLFolderItemSelectorView.java:251]: @W2jx, the target seems be incorrect. It should be https://github.com/liferay-core-infra/liferay-portal/blob/45553a42bea8cd0122d27af03ca78d029b6ab6ab/portal-impl/src/com/liferay/portlet/documentlibrary/service/permission/DLFolderPermission.java#L33

Comment (W2jx) [reply modules/apps/document-library/document-library-item-selector-web/src/main/java/com/liferay/document/library/item/selector/web/internal/folder/DLFolderItemSelectorView.java:251]: Hi @yuhai , The reason why I use `model.class.name=com.liferay.portal.kernel.repository.model.Folder`  is that the method 
```
public static boolean contains(
			PermissionChecker permissionChecker, long groupId, long folderId,
			String actionId)
```
in class DLFolderPermission uses _folderModelResourcePermission.

**Comment by W2jx** [comment]
ci:test:sf

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
ci:test:document

**Comment by zhangyan0701** [comment]
No regressions found.

**Comment by W2jx** [comment]
no conflicts(6/5 7:43)

**Comment by W2jx** [comment]
only modify the commit  message, no need to rerun test

**Comment by W2jx** [comment]
Hi @yuhai , I have moved the class GroupModelResourcePermission to the module site-impl.

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
ci:test:sf

**Comment by W2jx** [comment]
ci:test:document

**Comment by catarineaguiar** [comment]
no regressions found.

**Comment by W2jx** [comment]
no conflicts(6/6 8:46)

**Review by yuhai — APPROVED** [review]
@dantewang, the pr is good for me. The commit https://github.com/liferay-core-infra/liferay-portal/pull/4685/commits/bc447d825735475ea3cf56cce3d05f3c21782ec5 move GroupModelResourcePermission to module. This is the suited module which we can find.

**Comment by dantewang** [comment]
@W2jx Please add the `GroupModelResourcePermission` directly to site-impl, as `LayoutModelResourcePermission` and `UserModelResourcePermission` are already in -impl modules.

**Comment by W2jx** [comment]
Hi @dantewang , I have added the `GroupModelResourcePermission` directly to site-impl.

**Comment by W2jx** [comment]
no logic changes, no need to rerun test

**Comment by W2jx** [comment]
no conflicts(6/7 8:27)

**Review by dantewang — APPROVED** [review]

**Comment by tinatian** [comment]
@W2jx I can not sent this to BChan as it has code conflicts with BChan's master, please wait for those changes to be merged into upstream then rebase this pull.

Thanks.
Tina

**Comment by W2jx** [comment]
Hi @tinatian , I have rebase this pull.

**Comment by W2jx** [comment]
Forced push to resolve the conflict about bnd file, no logic changes.

**Comment by tinatian** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/136115

**Pull request closed without merging**